src/main/java/org/assertj/core/internal/Strings.java

```diff
@@ -330,6 +331,47 @@ private boolean areEqualIgnoringCase(CharSequence actual, CharSequence expected)
   }
 
   /**
+   * Verifies that two {@code CharSequence}s are equal, ignoring any changes in whitespace.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the actual {@code CharSequence}.
+   * @param expected the expected {@code CharSequence}.
+   * @throws AssertionError if the given {@code CharSequence}s are not equal.
+   */
+  public void assertEqualsIgnoringWhitespace(AssertionInfo info, CharSequence actual, CharSequence expected) {
+      if (!areEqualIgnoringWhitespace(actual, expected)) {
+        throw failures.failure(info, shouldBeEqual(actual, expected));
+      }
+  }
+
+  private boolean areEqualIgnoringWhitespace(CharSequence actual, CharSequence expected) {
+      if (actual == null) {
+          return expected == null;
+      }
+
+
+      return stripSpace(actual).equals(stripSpace(expected));
+  }
+
+  private String stripSpace(CharSequence toBeStripped) {
```

Comment:
why not just remove all whitespaces ? this will simplify this method.
if I'm right, we should rename it to `removeAllWhitespaces`